backend/extractors/tram_classifier_extractor.py

```diff
@@ -0,0 +1,127 @@
+import torch
+import json
+import joblib
+from transformers import BertTokenizer
+from typing import List, Optional
+from huggingface_hub import hf_hub_download
+import blingfire
+from collections import defaultdict
+
+class mitreClassifierExtractor:
+    def __init__(
+        self,
+        article_content: str,
+        model_repo: str = "dvir056/mitre_ttp",
+        threshold: float = 0.3,
+        keywords: List[str] = [],
+        qna: Optional[List[dict]] = None,
+    ):
+        self.article_content = article_content
+        self.threshold = threshold
+        self.keywords = keywords
+        self.qna = qna or []
+        self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+        # === Load model, tokenizer, label binarizer from Hugging Face ===
+        self.model_path = model_repo
+        self.tokenizer = BertTokenizer.from_pretrained(self.model_path)
+
+        scripted_model_path = hf_hub_download(repo_id=model_repo, filename="model_scripted.pt")
+        self.model = torch.jit.load(scripted_model_path).to(self.device)
+        self.model.eval()
+
+        label_binarizer_path = hf_hub_download(repo_id=model_repo, filename="label_binarizer.pkl")
+        self.mlb = joblib.load(label_binarizer_path)
+
+        mitre_json_path = hf_hub_download(repo_id=model_repo, filename="enterprise-attack.json")
+        self.mitre_map = self._load_mitre_map(mitre_json_path)
+
+    def _load_mitre_map(self, json_path: str):
+        with open(json_path, "r", encoding="utf-8") as f:
+            data = json.load(f)
+
+        mapping = {}
+        for obj in data["objects"]:
+            if obj.get("type") == "attack-pattern":
+                for ref in obj.get("external_references", []):
+                    if ref.get("source_name") == "mitre-attack" and "external_id" in ref:
+                        mapping[ref["external_id"]] = {
+                            "name": obj.get("name", "Unknown"),
+                            "url": ref.get("url", "")
+                        }
+        return mapping
+
+    def classify(self):
+        chunks = []
+
+        # 1. Split article
+        if len(self.tokenizer.encode(self.article_content, add_special_tokens=False)) > 512:
```

Comment:
if the model config is available, you can get the 512 from it instead of hardcoding.. (usually there is a model config json file)